buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -59,22 +102,282 @@ public Distribution getDistribution() {
+        return javaHome;
+    }
+
+    private void waitForUri(String description, String uri) {
+        waitConditions.put(description, (node) -> {
+            try {
+                URL url = new URL("http://" + this.getHttpPortInternal().get(0) + uri);
+                HttpURLConnection con = (HttpURLConnection) url.openConnection();
+                con.setRequestMethod("GET");
+                con.setConnectTimeout(500);
+                con.setReadTimeout(500);
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(con.getInputStream()))) {
+                    String response = reader.lines().collect(Collectors.joining("\n"));
+                    logger.info("{} -> {} ->\n{}", this, uri, response);
+                }
+                return true;
+            } catch (IOException e) {
+                throw new TestClustersException("Connection attempt to " + this + " failed", e);
+            }
+        });
+    }
+
+    synchronized void start() {
         logger.info("Starting `{}`", this);
+
+        File distroArtifact = new File(
+            new File(artifactsExtractDir, distribution.getFileExtension()),
+            distribution.getFileName() + "-" + getVersion()
+        );
+        if (distroArtifact.exists() == false) {
+            throw new TestClustersException("Can not start " + this + ", missing: " + distroArtifact);
+        }
+        if (distroArtifact.isDirectory() == false) {
+            throw new TestClustersException("Can not start " + this + ", is not a directory: " + distroArtifact);
+        }
+        services.sync(spec -> {
+            spec.from(new File(distroArtifact, "config"));
+            spec.into(getConfigFile().getParent());
+        });
+        writeConfigFile();
+        startElasticsearchProcess(distroArtifact);
+        registerCleanupHooks();
     }
 
-    void stop(boolean tailLogs) {
+    private void startElasticsearchProcess(File distroArtifact) {
+        logger.info("Running `bin/elasticsearch` in `{}` for {}", workingDir, this);
+        final ProcessBuilder processBuilder = new ProcessBuilder();
+        if (OperatingSystem.current().isWindows()) {
+            processBuilder.command(
+                "cmd", "/c",
+                new File(distroArtifact, "\\bin\\elasticsearch.bat").getAbsolutePath()
+            );
+        } else {
+            processBuilder.command(
+                new File(distroArtifact.getAbsolutePath(), "bin/elasticsearch").getAbsolutePath()
+            );
+        }
+        try {
+            processBuilder.directory(workingDir);
+            Map<String, String> environment = processBuilder.environment();
+            // Don't inherit anything from the environment for as that would  lack reproductability
+            environment.clear();
+            if (javaHome != null) {
+                environment.put("JAVA_HOME", getJavaHome().getAbsolutePath());
+            } else if (System.getenv().get("JAVA_HOME") != null) {
```

Comment:
Wouldn't it be better to get this from the project like we do currently (`project.runtimeJavaHome`)?